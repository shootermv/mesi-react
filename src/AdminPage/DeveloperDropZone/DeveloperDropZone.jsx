import React from 'react';


class DeveloperDropZone extends React.Component {

    onDragOver = (e) => {	
        e.stopPropagation();	
        e.preventDefault();	
    }	

    onDragEnter = (e) => {	
        e.stopPropagation();	
    }	

    render() {
        const { user, assignTaskToUser } = this.props;
        return (
            <div className='col-xs-4' 
            onDragEnter={this.onDragEnter}
            onDragOver={this.onDragOver}
            onDrop={e => assignTaskToUser(user, e)}>
               <div>{user.firstName + ' ' + user.lastName}</div>
               <div className='well'>
               {user.tasks && user.tasks.length ? user.tasks.map(({_id, summary, status}) => <li key={_id} draggable="true">{summary} {status}</li>):'no tasks yet'}
               </div>
            </div>
        );
    }
}



export  {DeveloperDropZone};