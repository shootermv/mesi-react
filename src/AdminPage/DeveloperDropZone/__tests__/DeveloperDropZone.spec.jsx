// test file
import React from 'react';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });
import {DeveloperDropZone}  from '../DeveloperDropZone'

test('Should render DeveloperDropZone component', () => {
    const testUser = {
        firstName: '',
        lastName: '',
        tasks: []
    }
    const wrapper = Enzyme.shallow(<DeveloperDropZone user={testUser} />);
    expect(wrapper).toMatchSnapshot();
});