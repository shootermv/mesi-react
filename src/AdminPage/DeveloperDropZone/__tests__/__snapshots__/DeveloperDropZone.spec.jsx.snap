// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DeveloperDropZone component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeveloperDropZone
    user={
      Object {
        "firstName": "",
        "lastName": "",
        "tasks": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
           
        </div>,
        <div
          className="well"
        >
          no tasks yet
        </div>,
      ],
      "className": "col-xs-4",
      "onDragEnter": [Function],
      "onDragOver": [Function],
      "onDrop": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
        },
        "ref": null,
        "rendered": " ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "no tasks yet",
          "className": "well",
        },
        "ref": null,
        "rendered": "no tasks yet",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
             
          </div>,
          <div
            className="well"
          >
            no tasks yet
          </div>,
        ],
        "className": "col-xs-4",
        "onDragEnter": [Function],
        "onDragOver": [Function],
        "onDrop": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "no tasks yet",
            "className": "well",
          },
          "ref": null,
          "rendered": "no tasks yet",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
